src/Training.cpp

```diff
@@ -82,10 +82,10 @@ std::istream& operator>> (std::istream& stream, TimeStep& timestep) {
     }
     int prob_size;
     stream >> prob_size;
-    for (auto i = 0; i < prob_size; ++i) {
+    for (auto i = 0; i < prob_size && i < POTENTIAL_MOVES; ++i) {
```

Comment:
Why does it need the extra condition?

Reply:
I am just checking for out-of-bound because I'm now accessing by index. (I don't know what I can assume about prob_size.) Anyway it's moot now that I'm switching back to vector.